packages/hardhat-verify/src/index.ts

```diff
@@ -282,11 +282,21 @@ subtask(TASK_VERIFY_VERIFY)
         throw new InvalidLibrariesError();
       }
 
-      await run(TASK_VERIFY_ETHERSCAN, {
-        address,
-        constructorArgsParams: constructorArguments,
-        libraries,
-        contract,
-      });
+      if (config.etherscan.enabled) {
```

Comment:
If the Etherscan verification fails, it will prevent the Sourcify verification from happening. @fvictorio, do you think we should mimic the main task's behavior and throw an error at the end?

Reply:
No, let's keep it this way. It's simpler, and I think that failing as soon as one of the providers fails makes more sense for programmatic verification.